src/main/java/com/endava/internship/collections/StudentList.java

```diff
@@ -211,26 +213,53 @@ public boolean addAll(Collection<? extends Student> collection) {
 
     @Override
     public boolean containsAll(Collection<?> collection) {
-        //Ignore this for homework
-        throw new UnsupportedOperationException();
+        for (Object e : collection)
+            if (!contains(e)) {
+                return false;
+            }
+        return true;
     }
 
     @Override
     public boolean addAll(int i, Collection<? extends Student> collection) {
-        //Ignore this for homework
-        throw new UnsupportedOperationException();
+        int counter = 0;
+        Iterator<? extends Student> it = collection.iterator();
+        while (it.hasNext()) {
+            Student student = it.next();
+            add(i + counter++, student);
+        }
+        return counter != 0;
     }
 
     @Override
     public boolean removeAll(Collection<?> collection) {
-        //Ignore this for homework
-        throw new UnsupportedOperationException();
+        Objects.requireNonNull(collection);
+        Iterator<?> it = collection.iterator();
+        while (it.hasNext()) {
+            Object student = it.next();
+            remove(student);
+        }
+        return !containsAll(collection);
     }
 
     @Override
     public boolean retainAll(Collection<?> collection) {
-        //Ignore this for homework
-        throw new UnsupportedOperationException();
+        Objects.requireNonNull(collection);
```

Comment:
This method can be refactored.
Collection is interface that has contains() method. 
We can use it in order to determine if student from the list contains in this collection, then we can remove this element.